### PR TITLE
add sync draft

### PR DESCRIPTION
**Pull request by maximsmol**

### CODE REVIEW

**Review by AidanAbd — APPROVED** [review]
This looks good to me.

**Review by AidanAbd — APPROVED** [review]

**Review by AidanAbd — APPROVED** [review]

File: latch_cli/main.py [hunk 1]
```diff
@@ -684,3 +684,37 @@ def test_data_ls():
...
+@click.option(
+    "--delete",
+    help="Delete extraneous files from destination.",
+    is_flag=True,
+    default=False,
+)
+@click.option(
+    "--ignore-unsyncable",
+    help=(
+        "Synchronize even if some source paths are do not exist or refer to special"
```
Comment (ayushkamat) [inline latch_cli/main.py:701]:
```suggestion
        "Synchronize even if some source paths do not exist or refer to special"
```

what is a special file

File: latch_cli/services/sync.py [hunk 2]
```diff
@@ -0,0 +1,328 @@
...
+    parts: List[upl.CompletedPart] = []
+    for idx, url in enumerate(start.urls):
+        parts.append(
+            upl.upload_file_chunk(
+                src,
+                url,
+                idx,
+                start.part_size,
+            )
+        )
```
Comment (ayushkamat) [inline latch_cli/services/sync.py:30]: why not upload in parallel?

File: latch_cli/services/sync.py [hunk 3]
```diff
@@ -0,0 +1,328 @@
...
+        if is_dir:
+            sub_srcs: Dict[str, Tuple[Path, os.stat_result]] = {}
+            for x in p.iterdir():
+                res = check_src(x, indent=indent + "  ")
+                if res is None:
+                    # todo(maximsmol): pre-check or confirm?
+                    continue
+
+                sub_srcs[x.name] = res
+            sync_rec(sub_srcs, child_dest, delete=delete, level=level + 1)
```
Comment (ayushkamat) [inline latch_cli/services/sync.py:263]: ideally id like to avoid doing a gql request in every recursive call, but this is fine for now

**Review by ayushkamat — CHANGES_REQUESTED** [review]

**Pull request merged**